ci/kind/test-secondary-network-kind.sh

```diff
@@ -0,0 +1,125 @@
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+# The script runs NetworkPolicy conformance tests on Kind with different traffic encapsulation modes.
+
+set -eo pipefail
+
+function echoerr {
+    >&2 echo "$@"
+}
+
+_usage="Usage: $0 [--setup-only|--test-only|--cleanup-only] [--help|-h]
+        --setup-only                  Only perform setting up the cluster and run test.
+        --test-only                   Only run test on current cluster. Not set up/clean up the cluster.
+        --cleanup-only                Only perform cleaning up the cluster.
+        --help, -h                    Print this message and exit.
+"
+
+function print_usage {
+    echoerr -n "$_usage"
+}
+
+TESTBED_CMD=$(dirname $0)"/kind-setup.sh"
+YML_CMD=$(dirname $0)"/../../hack/generate-manifest.sh"
+ATTACHMENT_DEFINITION_YAML=$(dirname $0)"/../../test/e2e-secondary-network/infra/network-attachment-definition-crd.yml"
+SECONDARY_NETWORKS_YAML=$(dirname $0)"/../../test/e2e-secondary-network/infra/secondary-networks.yml"
+
+TIMEOUT="5m"
+# Create a secondary OVS bridge with the Node's physical interface on the extra
+# network.
+MANIFEST_ARGS="--secondary-bridge br-secondary --physical-interface eth1"
+# Antrea is deployed by this script. Do not deploy it again in the test.
+TEST_OPTIONS="--logs-export-dir=$ANTREA_LOG_DIR --deploy-antrea=false"
+EXTRA_NETWORK="20.20.20.0/24"
+
+setup_only=false
+cleanup_only=false
+test_only=false
+
+function quit {
+  result=$?
+  if [[ $setup_only || $test_only ]]; then
+    exit $result
+  fi
+  echoerr "Cleaning testbed"
+  $TESTBED_CMD destroy kind
+}
+
+while [[ $# -gt 0 ]]
+do
+key="$1"
+
+case $key in
+    --setup-only)
+    setup_only=true
+    shift
+    ;;
+    --cleanup-only)
+    cleanup_only=true
+    shift
+    ;;
+    --test-only)
+    test_only=true
+    shift
+    ;;
+    -h|--help)
+    print_usage
+    exit 0
+    ;;
+    *)    # unknown option
+    echoerr "Unknown option $1"
+    exit 1
+    ;;
+esac
+done
+
+if [[ $cleanup_only == "true" ]];then
+  $TESTBED_CMD destroy kind
+  exit 0
+fi
+
+trap "quit" INT EXIT
+
+IMAGE_LIST=("projects.registry.vmware.com/antrea/busybox" \
+            "antrea/antrea-ubuntu:latest")
+
+printf -v IMAGES "%s " "${IMAGE_LIST[@]}"
+
+function setup_cluster {
+  args=$1
+  echo "creating test bed with args $args"
+  eval "timeout 600 $TESTBED_CMD create kind $args"
+}
+
+function run_test {
+  echo "Generating Antrea manifest with args $MANIFEST_ARGS"
+  $YML_CMD $MANIFEST_ARGS | kubectl apply -f -
```

Comment:
I would still prefer to keep the generate-manifest.sh changes, as it is simpler to run generate-manifest.sh then using the helm command. But I can change this test script to use "helm install" if you want that.

But could you let me know the best way to pass the ovsBridges parameter to helm? I tried `secondaryNetwork.ovsBridges="[{bridgeName: br-secondary, physicalInterfaces: [eth1]}]"` and it does not work.

Reply:
You can use:
```
helm template antrea build/charts/antrea/ --set featureGates.SecondaryNetwork=true --set-json secondaryNetwork.ovsBridges='[{"bridgeName": "br-secondary", "physicalInterfaces": ["eth1"]}]'
```

I don't think there is a YAML equivalent to `--set-json`. I doubt it because formatting is important in YAML and part of the syntax.